src/mixing_fab_tests.cc

```diff
@@ -0,0 +1,386 @@
+  
+  
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+  
+  Composition::Ptr c_pustream() {
+    CompMap m;
+    m[id("pu239")] = 100;
+    m[id("pu240")] = 10;
+    m[id("pu241")] = 1;
+    m[id("pu242")] = 1;
+    return Composition::CreateFromMass(m);
+  };
+  
+  Composition::Ptr c_natu() {
+    CompMap m;
+    m[id("u235")] =  .007;
+    m[id("u238")] =  .993;
+    return Composition::CreateFromMass(m);
+  };
+  
+  Composition::Ptr c_uox() {
+    CompMap m;
+    m[id("u235")] = 0.04;
+    m[id("u238")] = 0.96;
+    return Composition::CreateFromMass(m);
+  };
+  
+  
+  class MixingFabTest : public ::testing::Test {
+    
+    
+  public:
+    typedef cyclus::toolkit::ResBuf<cyclus::Material> InvBuffer;
+    cyclus::TestContext tc_;
+    MixingFab* mf_facility_;
+    
+    virtual void SetUp();
+    virtual void TearDown();
+    
+    std::vector<std::string> in_com;
+    std::vector<double> in_frac;
+    std::vector<double> in_cap;
+    
+    
+    std::string out_com;
+    double out_cap;
+    
+    double throughput;
+    
+    void SetThroughput(double thpt) { throughput = thpt; mf_facility_->throughput = thpt;}
+    
+    void SetStream_comds(std::vector<std::string> comds)   { in_com  = comds;  mf_facility_->in_commods   = comds; }
+    void SetStream_ratio(std::vector<double> ratio)        { in_frac = ratio;  mf_facility_->mixing_ratio = ratio; }
+    void SetStream_capacity(std::vector<double> cap)       { in_cap  = cap;    mf_facility_->in_buf_size  = cap; }
+    
+    void SetOutStream_comds(std::string com)  { out_com = com; mf_facility_->out_commod      = com; }
+    void SetOutStream_capacity(double cap)    { out_cap = cap; mf_facility_->output_buf_size = cap; }
+    void SetInputInv(std::vector<Material::Ptr> mat);
+    
+    
+    std::vector<std::string>  GetStream_comds() { return mf_facility_->in_commods; }
+    std::vector<double> GetStream_ratio()       { return mf_facility_->mixing_ratio; }
+    std::vector<double> GetStream_capacity()    { return mf_facility_->in_buf_size; }
+    
+    std::string GetOutStream_comds()  { return mf_facility_->out_commod; }
+    double GetOutStream_capacity()    { return mf_facility_->output_buf_size; }
+    double GetThroughput()    { return mf_facility_->throughput; }
+    
+    
+    InvBuffer* GetOutPutBuffer() { return &mf_facility_->output;}
+    std::map<std::string, InvBuffer > GetStreamBuffer() { return mf_facility_->streambufs;}
+    
+    
+    
+    
+  };
+  
+  
+  void MixingFabTest::SetUp() {
+    mf_facility_ = new MixingFab(tc_.get());
+ 
+  }
+  
+  void MixingFabTest::TearDown() {
+    delete mf_facility_;
+  }
+
+  
+  void MixingFabTest::SetInputInv(std::vector< Material::Ptr > mat){
+    for (int i = 0; i < in_com.size(); i++) {
+      mf_facility_->streambufs[in_com[i]].Push(mat[i]) ;
+    }
+  }
+  
+  
+  
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+  TEST_F(MixingFabTest, InitialState) {
```

Comment:
This test is just testing your test code - not the actual mixing facility.  I don't think it adds much and can probably just be removed.